src/auth.d.ts

```diff
@@ -0,0 +1,1581 @@
+import * as _admin from './index.d';
```

Comment:
Isn't this creating a circular dependency? Maybe a separate file for App and Error should be created? Similarly in the other files.

Reply:
It looks like TS allows this kind of circular dependencies in typings. It's really not possible to avoid it in our API, since our `App` type has methods like `auth()` and `database()` that returns `admin.auth.Auth` and `admin.database.Database` respectively. So we essentially have this dependency chain:

* `admin.auth.Auth` references `admin.app.App`.
* `admin.app.App` has a method `auth()` that returns `admin.auth.Auth`.

Same applies to all other modules.